### PR TITLE
Public api initial endpoint and setup

### PR DESCRIPTION
**Pull request by RubenSousaDinis**
## Summary

This PR adds the initial API structure.

- The first endpoint is being added `/profiles/id`
- Docs are already being generated

How to generate API docs:

`RAILS_ENV=test SWAGGER_DRY_RUN=0 be rails rswag`

## Notion link

https://www.notion.so/talentprotocol/Lens-Integration-API-endpoint-7adf2e69f54f44d39862272a30144fc0?pvs=4

### CODE REVIEW

File: app/controllers/api/v1/public_api/profiles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Inherit from this controller for authenticated api requests
```
Comment (verdefred) [inline app/controllers/api/v1/public_api/profiles_controller.rb:1]: this comment only makes sense in the other controller, right?

**Review by helloiambguedes — APPROVED** [review]

**Review by verdefred — COMMENTED** [review]
Great work, just left a couple of comments

File: app/controllers/api/v1/public_api/profiles_controller.rb (hunk 1 above)
Comment (RubenSousaDinis) [reply app/controllers/api/v1/public_api/profiles_controller.rb:1]: Right!

File: db/migrate/20230220164338_create_api_keys.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateAPIKeys < ActiveRecord::Migration[7.0]
+  def change
+    create_table :api_keys do |t|
+      t.text :access_key, null: false
+      t.string :name, null: false
+      t.string :description
+      t.timestamp :activated_at
+      t.timestamp :revoked_at
```
[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply db/migrate/20230220164338_create_api_keys.rb:8]: Yes, we can have it

[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply db/migrate/20230220164338_create_api_keys.rb:8]: added

**Review by verdefred — APPROVED** [review]

**Pull request merged**